node/src/tasks/bsp_upload_file.rs

```diff
@@ -250,50 +250,79 @@ where
             }
         };
 
-        // Query runtime for the chunks to prove for the file.
-        let chunks_to_prove = self
+        let own_bsp_id = self
             .storage_hub_handler
             .blockchain
-            .query_bsp_confirm_chunks_to_prove_for_file(
-                self.storage_hub_handler
-                    .blockchain
-                    .get_node_public_key()
-                    .await,
-                event.file_key,
-            )
-            .await
-            .map_err(|e| {
-                anyhow!(
-                    "Failed to query BSP confirm chunks to prove for file: {:?}",
-                    e
-                )
-            })?;
-
-        // Generate the proof for the file.
+            .get_node_public_key()
+            .await;
+
+        // Query runtime for the chunks to prove for the file.
+        let mut file_key_with_chunks_to_prove = Vec::new();
+        for file_key in event.file_keys.iter() {
+            match self
+                .storage_hub_handler
+                .blockchain
+                .query_bsp_confirm_chunks_to_prove_for_file(own_bsp_id, *file_key)
+                .await
+            {
+                Ok(chunks_to_prove) => {
+                    file_key_with_chunks_to_prove.push((*file_key, chunks_to_prove));
+                }
+                Err(e) => {
+                    error!(target: LOG_TARGET, "Failed to query chunks to prove for file {:?}: {:?}\nEnqueuing file key again!", file_key, e);
+                    self.storage_hub_handler
+                        .blockchain
+                        .queue_confirm_bsp_request(ConfirmStoringRequest::new(*file_key))
```

Comment:
This is a fair strategy, but could lead to something always being there in the queue infinitely, if this is a consistent error. I think for this particular case, if this fails and we fail to confirm a file, it is not that bad to just give up that file. Let me know your thoughts.